tests/_expected_data_for_fragmentation_test.py

```diff
@@ -828,6 +828,16 @@ def get_expected(key: str):
                     23,
                     25,
                     27,
+                    33,
+                    34,
+                    35,
+                    36,
+                    37,
+                    14,
+                    15,
+                    16,
+                    17,
+                    18,
```

Comment:
These changes in the expected data confuse me. Does the code currently generate incorrect fragmentation? If so, which cases were affected?